tests/scripts/all.sh

```diff
@@ -899,6 +899,33 @@ component_build_deprecated () {
     make CC=clang CFLAGS='-O -Werror -Wall -Wextra -Wno-unused-function' tests
 }
 
+# Check that the specified libraries exist and are empty.
+are_empty_libraries () {
+  nm "$@" >/dev/null 2>/dev/null
+  ! nm "$@" 2>/dev/null | grep -v ':$' | grep .
```

Comment:
Very minor: I don't think the final `| grep .` is necessary here, `grep -v` should already return 0 if lines were selected (that don't match the pattern): https://pubs.opengroup.org/onlinepubs/9699919799/utilities/grep.html It clearly doesn't hurt, though.

Reply:
There are empty lines as well. Those are acceptable.

Reply:
Oh, good point, I missed that.